lib/xenops_server_simulator.ml

```diff
@@ -142,7 +145,11 @@ let get_domain_action_request_nolock vm () =
 let destroy_nolock vm () =
   debug "Domain.destroy vm=%s" vm.Vm.id ;
   (* Idempotent *)
```

Comment:
This isn't idempotent anymore so we should remove this comment (f . f  x is not equivalent to f x if you resume in between)

Reply:
I don't think that allowing another function in between the f's is required for idempotency?